src/zenml/client.py

```diff
@@ -3822,23 +3880,29 @@ def _get_entity_by_prefix(
             ZenKeyError: If there is more than one entity with that partial ID
                 or name.
         """
-        entity = list_method(
-            logical_operator=LogicalOperators.OR,
-            name=f"contains:{partial_id_or_name}",
-            id=f"startswith:{partial_id_or_name}",
-        )
+        list_method_args: Dict[str, Any] = {
+            "logical_operator": LogicalOperators.OR,
+            "id": f"startswith:{partial_id_or_name}",
+        }
+        if allow_name_prefix_match:
+            list_method_args["name"] = f"startswith:{partial_id_or_name}"
```

Comment:
```suggestion
        if allow_name_prefix_match:
            list_method_args["name"] = f"startswith:{partial_id_or_name}"
        else:
            list_method_args["name"] = partial_id_or_name        
```

I think you missed this

Reply:
As far as I understood, that case is already handled in the method that calls this, which covers both the exact ID and exact name match

Reply:
Ah, of course. You're right.